browser.js

```diff
@@ -14,11 +14,14 @@ if (crypto && crypto.getRandomValues) {
 
 function randomBytes (size, cb) {
   // in browserify, this is becomes an extended UInt8Array
-  var bytes = new Buffer(size)
+  var rawBytes = new global.UInt8Array(size)
 
   // This will not work in older browsers.
   // See https://developer.mozilla.org/en-US/docs/Web/API/window.crypto.getRandomValues
-  crypto.getRandomValues(bytes)
+  crypto.getRandomValues(rawBytes)
```

Comment:
In browsers that support Typed Arrays *and* have a correct implementation, `Buffer` will be a subclass of `Uint8Array`. In older or broken browsers, `Buffer` will just be an `Object`. You can see the exactly support list here: https://github.com/feross/buffer/blob/master/index.js#L20-L49

It's my understanding that `crypto.getRandomValues` absolutely must be given a `Uint8Array`, so it's expected to fail in browsers without support. Does the version of phantomjs you're using have support?

You can try logging out the value of `Buffer.TYPED_ARRAY_SUPPORT` to see if Buffer is actually using the `Uint8Array` implementation.